Application/EdFi.Ods.Tests/EdFi.Ods.Api/Controllers/TokenControllerUsingJsonTests.cs

```diff
@@ -68,55 +176,119 @@ protected override Task ArrangeAsync()
                             new ApiClientAuthenticator.AuthenticationResult
                             {
                                 IsAuthenticated = true,
-                                ApiClientIdentity = new ApiClientIdentity {Key = "clientId"}
+                                ApiClientIdentity = new ApiClientIdentity { Key = "clientId" }
                             }));
 
-                var _tokenRequestProvider = Stub<ClientCredentialsTokenRequestProvider>();
+                _controller = ControllerHelper.CreateTokenController(_clientAppRepo, _apiClientAuthenticator);
+
+                return Task.CompletedTask;
+            }
+
+            protected override async Task ActAsync()
+            {
+                _controller.ControllerContext = CreateControllerContext(
+                    new HeaderDictionary { { "Authorization", $"Basic {CreateEncodedAuthentication()}" } });
```

Comment:
This should be part of the CreateTokenController(). My suggestion is to make this as a parameter to the method with a default of of the bearer token, and override as a accordingly.

Reply:
CreateControllerContext is getting called in ActAsync() but CreateTokenController() is in ArrangeAsync() since both are two indipendent methods how can we merge them?